tests/unit/test_utils_filesystem.py

```diff
@@ -59,3 +70,53 @@ def test_copy2_fixed_raises_appropriate_errors(create, error_type, tmpdir):
         copy2_fixed(src, dest)
 
     assert not dest.exists()
+
+
+def test_file_opener_no_file(process):
+    # FileOpener joins the subprocess even if the parent never sends the path
+    with FileOpener():
+        pass
+    assert len(process.children()) == 0
+
+
+def test_file_opener_not_found(tmpdir, process):
+    # The FileOpener cleans up the subprocess when the file cannot be opened
+    path = tmpdir.joinpath('foo.txt')
+    with FileOpener(path):
+        pass
+    assert len(process.children()) == 0
+
+
+def test_file_opener_normal(tmpdir, process):
+    # The FileOpener cleans up the subprocess when the file exists
+    path = tmpdir.joinpath('foo.txt')
+    with open(path, 'w') as f:
+        f.write('Hello\n')
+    with FileOpener(path):
+        pass
+    assert len(process.children()) == 0
+
+
+@skip_unless_windows
+def test_file_opener_produces_unlink_error(tmpdir, process):
+    # FileOpener forces an error on Windows when we attempt to remove a file
+    # The initial path may be deferred; which must be tested with an error
+    path = tmpdir.joinpath('foo.txt')
+    with open(path, 'w') as f:
+        f.write('Hello\n')
+    with FileOpener() as opener:
+        opener.send(path)
+        with pytest.raises(OSError):
```

Comment:
Can we assert facts about this error?

Reply:
Yes, we can insist it is a EACCES error.  Part of the point of my patch to pip/_internal/utils/temp_dir.py, is that it doesn't matter whether it is in fact windows error 5 or 32, because we know this is a temporary directory we just created.

Reply:
Done